dbms/src/DataStreams/ConvertingBlockInputStream.cpp

```diff
@@ -100,9 +109,18 @@ Block ConvertingBlockInputStream::readImpl()
     Block res = header.cloneEmpty();
     for (size_t res_pos = 0, size = conversion.size(); res_pos < size; ++res_pos)
     {
-        const auto & src_elem = src.getByPosition(conversion[res_pos]);
         auto & res_elem = res.getByPosition(res_pos);
 
+        if (conversion[res_pos] == USE_DEFAULT)
+        {
+            // Create a column with default values
+            auto column_with_defaults = res_elem.type->createColumn()->cloneResized(src.rows());
```

Comment:
Will it calculate DEFAULT expressions? Can you add a test for that?

Reply:
Yes, there is an issue with calculating default expressions. The test below shows that it only gets calculated for the first row.